.github/workflows/gh-build-and-test.yml

```diff
@@ -0,0 +1,71 @@
+on:
+  workflow_call:
+    inputs:
+      device:
+        type: string
+        required: true
+      build-runner:
+        type: string
+        required: true
+      platform:
+        type: string
+        required: true
+
+
+jobs:
+  build:
+    if: ${{ github.repository_owner == 'nv-legate' }}
+    uses:
+      ./.github/workflows/gh-build.yml
+    with:
+      build-target: ${{ inputs.device }}
+      repos-name: ${{ github.event.repository.name }}
+      runs-on: ${{ inputs.build-runner }}
+      sha: ${{ github.sha }}
+      build-type: ci
+      use-container: ${{ inputs.platform == 'linux' }}
+
+  test:
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Pytest Unit Tests
+            test-scope: unit
+            runner: linux-amd64-gpu-v100-latest-1
+            enabled: ${{ inputs.platform == 'linux' }}
+
+          - name: Pytest Unit Tests (OS X)
+            test-scope: unit
+            runner: macos-latest
+            enabled: ${{ inputs.platform == 'osx' }}
+
+          - name: Pytest Unit Tests
+            test-scope: unit
+            runner: linux-amd64-2gpu
+            enabled: ${{ inputs.platform == 'linux' }}
+
+          - name: mypy
+            test-scope: mypy
+            runner: linux-amd64-gpu-v100-latest-1
+            enabled: ${{ inputs.platform == 'linux' }}
+
+          # - name: mypy (OS X)
```

Comment:
Do we have someone who's following up with the things mypy is flagging?

Reply:
I have asked @manopapad to help with this.

Reply:
It's fixed now.